### ISSUE TITLE
configure.py fails on non-English Windows

### ISSUE THREAD

**Issue by securitykernel** [randombit/botan#1125]
Running configure.py on a German Windows 10 with Visual Studio 2017 fails in the compiler version detection.

```
botan>py.exe configure.py
   INFO: configure.py invoked with options ""
   INFO: Platform: OS="Windows" machine="AMD64" proc="Intel64 Family 6 Model 94 Stepping 3, GenuineIntel"
   INFO: Guessing target OS is windows (use --os to set)
   INFO: Guessing to use compiler msvc (use --cc to set)
   INFO: Guessing target processor is a x86_64/x86_64 (use --cpu to set)
   INFO: Canonicalized CPU target x86_64 to x86_64/x86_64
   INFO: Target is msvc-windows-x86_64-x86_64
Exception in thread Thread-2:
Traceback (most recent call last):
  File "C:\Program Files\Python36\lib\threading.py", line 916, in _bootstrap_inner
    self.run()
  File "C:\Program Files\Python36\lib\threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Program Files\Python36\lib\subprocess.py", line 1055, in _readerthread
    buffer.append(fh.read())
  File "C:\Program Files\Python36\lib\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x81 in position 62: character maps to <undefined>

  ERROR: Traceback (most recent call last):
  File "configure.py", line 3032, in <module>
    sys.exit(main(argv=sys.argv))
  File "configure.py", line 3016, in main
    cc_version = detect_compiler_version(cc, options.compiler_binary or cc.binary_name, osinfo.basename)
  File "configure.py", line 2599, in detect_compiler_version
    universal_newlines=True).communicate()
  File "C:\Program Files\Python36\lib\subprocess.py", line 838, in communicate
    stdout, stderr = self._communicate(input, endtime, timeout)
  File "C:\Program Files\Python36\lib\subprocess.py", line 1107, in _communicate
    stderr = stderr[0]
IndexError: list index out of range

An internal error occurred.

Don't panic, this is probably not your fault!

Please report the entire output at https://github.com/randombit/botan or email
to the mailing list https://lists.randombit.net/mailman/listinfo/botan-devel

You'll meet friendly people happy to help!
```

Byte 0x81 is the German umlaut "ü". Digging further into the code I found that Botan invokes `cl.exe` and parses the output. On a German Windows installation the output is:

```
botan>cl.exe
Microsoft (R) C/C++-Optimierungscompiler Version 19.10.25019 für x86
Copyright (C) Microsoft Corporation. Alle Rechte vorbehalten.

Syntax: cl [ Option... ] Dateiname... [ /link Linkeroption... ]
```

So configure.py chokes on the "ü" in "für x86". Apart from the text encoding problem we may have a more serious issue here, because the regex in `detect_compiler_version()` that attempts to extract the MSVC version also requires the output of `cl.exe` to be English: `r'Compiler Version ([0-9]+).([0-9]+).[0-9\.]+ for'`.

**Comment by randombit** [comment]
Shakes fist at MSVC. All I want is the equivalent of `gcc -dumpversion` damn it.

Maybe we can get away with changing the regex to `r'Version ([0-9]+).([0-9]+).[0-9\.]+'` ... but that probably breaks for Chinese or etc. (Then again in your output 'Version' seems to have been left untranslated, so maybe we can rely on it being that regardless of locale?)

The good news is this method of compiler detection at build time is just a stopgap. We need it for feature detection (eg, checking if the compiler supports x86 SHA intrinsics). I hope to replace it with a system that just compiles generated test programs as mentioned in #866. At which point this troublesome logic can be killed off entirely.

**Comment by webmaster128** [comment]
MSVC has a very special output encoding, that needs to be handled separatly and explicitly. You can assign this ticket to me, I did this kind of stuff a lot for [clcache](https://github.com/frerich/clcache)

**Comment by randombit** [comment]
@webmaster128 That would be great thank you.

**Comment by randombit** [comment]
Hm for whatever reason GH is only allowing me to assign this to myself or @securitykernel, but anyway it's all yours.

**Comment by securitykernel** [comment]
@webmaster128 Thanks for taking up on this!

**Comment by webmaster128** [comment]
Created a fix in #1133.

One site note regarding this issue's title: the output language of cl.exe is not determined by the Windows language but by the installed version of Visual Studio. A default MSVS installation on a German Windows will be English. I had to install a language pack for Visual Studio to reproduce this.

**Comment by securitykernel** [comment]
The fix in #1133 works fine for me, thank you!

Interesting, I switched VS 2017 to English language after installing it in German and still cl.exe output is in German.
